models/magiceden/solana/magiceden_solana_trades.sql

```diff
@@ -0,0 +1,243 @@
+{{
+    config(
+        schema = 'magiceden_solana'
+        , tags = ['dunesql']
+        , alias = alias('trades')
+        ,materialized = 'incremental'
+        ,file_format = 'delta'
+        ,incremental_strategy = 'merge'
+        ,unique_key = ['project','trade_category','outer_instruction_index','inner_instruction_index','account_metadata','tx_id']
+        ,post_hook='{{ expose_spells(\'["solana"]\',
+                                    "project",
+                                    "magiceden",
+                                    \'["ilemi"]\') }}'
+    )
+}}
+
+with 
+    royalty_logs as (
+        with nested_logs as (
+            SELECT 
+                distinct 
+                call_tx_id
+                , call_block_slot
+                , call_outer_instruction_index
+                , call_inner_instruction_index
+                , cast(json_extract_scalar(json_parse(split(logs, ' ')[3]), '$.royalty') as double) as royalty
+                , logs
+            FROM (
+                SELECT
+                    call_tx_id
+                    , call_block_slot
+                    , call_outer_instruction_index
+                    , call_inner_instruction_index
+                    , call_log_messages
+                FROM {{ source('magic_eden_solana','m2_call_mip1ExecuteSaleV2') }}
+                UNION ALL 
+                SELECT
+                    call_tx_id
+                    , call_block_slot
+                    , call_outer_instruction_index
+                    , call_inner_instruction_index
+                    , call_log_messages
+                FROM {{ source('magic_eden_solana','m2_call_executeSaleV2') }}
+                UNION ALL 
+                SELECT
+                    call_tx_id
+                    , call_block_slot
+                    , call_outer_instruction_index
+                    , call_inner_instruction_index
+                    , call_log_messages
+                FROM {{ source('magic_eden_solana','m2_call_ocpExecuteSaleV2') }}
```

Comment:
same idea here. it's best to apply incremental filters directly on read in CTEs, rather than outside of CTEs